### PR TITLE
Fixing bugs with withdraw and adding gas estimation

### PR DESCRIPTION
**Pull request by IanWoodard**
Currently, the withdraw modal doesn't handle small values properly because they are sometimes represented in scientific notation. To remedy this, I updated the logic to utilize `Big`'s whenever possible, and when we converted them to strings, I made sure we used `.toFixed()`. Additionally, I added proper gas estimation rather than relying on hardcoded values.

### CODE REVIEW

File: earn/src/components/portfolio/modal/WithdrawModal.tsx
Comment (IanWoodard) [inline earn/src/components/portfolio/modal/WithdrawModal.tsx:250]: Big's can handle scientific notation, but BigNumber's cannot. That is why this is okay.

File: earn/src/components/portfolio/modal/WithdrawModal.tsx
Comment (haydenshively) [inline earn/src/components/portfolio/modal/WithdrawModal.tsx:91]: Since we're technically redeeming by calling the `redeem` function (not withdrawing), let's call this something like `numberOfSharesToRedeem`

File: earn/src/components/portfolio/modal/WithdrawModal.tsx
Comment (haydenshively) [inline earn/src/components/portfolio/modal/WithdrawModal.tsx:93]: Maybe `redeemConfig`?

File: earn/src/components/portfolio/modal/WithdrawModal.tsx
Comment (haydenshively) [inline earn/src/components/portfolio/modal/WithdrawModal.tsx:280]:
```suggestion
                  setInputValue(bigMaxWithdraw.div(10 ** selectedOption.decimals).toFixed(selectedOption.decimals));
```
We should always be explicit with this.

**Review by haydenshively — DISMISSED** [review]
A few nits but don't wait for another review before merging.

**Pull request merged**
